pdata/plog/generated_scopelogs.go

```diff
@@ -28,34 +28,60 @@ import (
 // This is a reference type, if passed by value and callee modifies it the
 // caller will see the modification.
 //
-// Must use NewScopeLogs function to create new instances.
+// Must use NewExemplar function to create new instances.
 // Important: zero-initialized instance is not valid for use.
 type ScopeLogs struct {
+	commonScopeLogs
+}
+
+type MutableScopeLogs struct {
+	commonScopeLogs
+	preventConversion struct{}
```

Comment:
What does this prevent? Want to learn this trick..

Reply:
it prevents the conversion between ScopeLogs MutableScopeLogs

Reply:
Sure, but people don't have access to `immutableScopeLogs` since is private. Who can do that conversion?

Reply:
By `immutableScopeLogs` I meant an instance of `ScopeLogs`

Reply:
Discussed this offline, resolved